src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -352,6 +369,8 @@ private void TokenizeUserData(JsonRpcMessage jsonRpcMessage)
             {
                 result.Result = this.TokenizeUserData(result.Result);
             }
+
+            this.requestId = null;
         }
```

Comment:
This should be in a finally block so we don't leave initialized fields around if we throw an exception.